output.go

```diff
@@ -0,0 +1,19 @@
+package shaping
+
+import (
+	"golang.org/x/image/font/sfnt"
+	"golang.org/x/image/math/fixed"
+)
+
+type Output interface {
+	// Advance returns the distance the Dot has advanced.
+	Advance() fixed.Int26_6
+	// Baseline returns the distance the baseline is from the top.
+	Baseline() fixed.Int26_6
+	// Bounds returns the smallest rectangle capable of containing the shaped text.
+	Bounds() fixed.Rectangle26_6
+	// Length returns the number of glyphs in the output.
+	Length() int
+	// Index returns the GlyphIndex to draw at the given index.
+	Index(int) sfnt.GlyphIndex
```

Comment:
I think the minimal information needed for a glyph is now:

- its index in the font (GlyphIndex) 
- its relation with the input characters: @khaledhosny has given a detailed explanation, but for now, I suggest we start simple and just use the Harfbuzz clusters (from which we can built an higher-level information)
- its position, so that it can later be drawn using the font

I would then propose the following structure for the Glyph type (copied from the Harfbuzz port):
```
type GlyphPosition struct {
	// How much the line advances after drawing this glyph when setting
	// text in horizontal direction.
	XAdvance Position
	// How much the glyph moves on the X-axis before drawing it, this
	// should not affect how much the line advances.
	XOffset Position

	// How much the line advances after drawing this glyph when setting
	// text in vertical direction.
	YAdvance Position
	// How much the glyph moves on the Y-axis before drawing it, this
	// should not affect how much the line advances.
	YOffset Position
}

type Glyph struct {
    Index sfnt.GlyphIndex // or later our own font.GID
    Cluster int // index in the input slice 
    Position GlyphPosition 
}
```